### PR TITLE
Add mbedtls library

### PR DESCRIPTION
**Pull request by mchindri**
- Add mbedtls library that will be used for tls sockets.
- Add config file to build mbedtls library
- Add dummy makefile of how mbedtls should be integrated

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Modify linux.mk (I think this is the right way to do it. I'am waiting for opinions)

**Comment by mchindri** [comment]
v3:
 - Modify config file with the one that works with secure tcp socket

File: tools/scripts/linux.mk [hunk 1]
```diff
@@ -379,6 +384,11 @@ ifeq (y,$(strip $(TINYIIOD)))
 	@$(MAKE) -C $(LIBRARIES)/libtinyiiod re
 LIBS += -L $(LIBRARIES)/libtinyiiod/build -ltinyiiod
 endif
+ifeq (y,$(strip $(MBEDTLS)))
+	@$(MAKE) -C $(LIBRARIES)/mbedtls clean
```
Comment (dbogdan) [inline tools/scripts/linux.mk:388]: Why do you want to do the clean every time?

File: tools/scripts/linux.mk [hunk 2]
```diff
@@ -96,6 +96,11 @@ CFLAGS += -D IIO_EXAMPLE
 CFLAGS += -D _USE_STD_INT_TYPES
 endif
 
+ifeq (y,$(strip $(MBEDTLS)))
+CFLAGS += -D MBEDTLS_CONFIG_FILE="""noos_mbedtls_config.h"""
```
Comment (dbogdan) [inline tools/scripts/linux.mk:100]: """noos_mbedtls_config.h""" -> "noos_mbedtls_config.h" ?

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,6 @@
 [submodule "libraries/libtinyiiod"]
```
Comment (adrimbarean) [inline .gitmodules:1]: Need signature on this commit.

File: network/transport/noos_mbedtls_config.h [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/***************************************************************************//**
```
Comment (adrimbarean) [inline network/transport/noos_mbedtls_config.h:1]: Need signature.

File: network/transport/noos_mbedtls_config.h [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+#define MBEDTLS_X509_CSR_PARSE_C
+#define MBEDTLS_X509_USE_C
+
+/* For test certificates */
+#define MBEDTLS_BASE64_C
+#define MBEDTLS_PEM_PARSE_C
+
+#include "mbedtls/check_config.h"
+
+#endif /* MBEDTLS_CONFIG_H */
```
Comment (adrimbarean) [inline network/transport/noos_mbedtls_config.h:89]: nitpick: no new line at end file

File: tools/scripts/linux.mk (hunk 1 above)
Comment (mchindri) [reply tools/scripts/linux.mk]: moved to clean

File: tools/scripts/linux.mk (hunk 2 above)
Comment (mchindri) [reply tools/scripts/linux.mk]: solved

File: .gitmodules (hunk 3 above)
Comment (mchindri) [reply .gitmodules:1]: done

File: network/transport/noos_mbedtls_config.h (hunk 5 above)
Comment (mchindri) [reply network/transport/noos_mbedtls_config.h:89]: solved

File: network/transport/noos_mbedtls_config.h (hunk 4 above)
Comment (mchindri) [reply network/transport/noos_mbedtls_config.h:1]: solved

**Comment by mchindri** [comment]
v4
- Add extra line
- Sign commits
- Move mbedtls clean to clean target
- Test make on linux and add fixes
- Change """noos_mbedtls_config.h"""->'"noos_mbedtls_config.h"'

**Comment by mchindri** [comment]
v5:
 - fix noos_mbedtls_config filename in file header
 - run mbedtls clean only if submodule exists

File: network/transport/noos_mbedtls_config.h (hunk 4 above)
Comment (mirceacaprioru) [reply network/transport/noos_mbedtls_config.h:1]: this still doesn't have a signature.

**Comment by mchindri** [comment]
v6
 - Sign 2nd commit

File: network/transport/noos_mbedtls_config.h (hunk 4 above)
Comment (mchindri) [reply network/transport/noos_mbedtls_config.h:1]: I missed this.
Solved now.

**Pull request merged**